### PR TITLE
Feature: Materialized record allow nil

### LINKED ISSUE #12: Support `allow_nil` for `materialised_record`

### PR DESCRIPTION
**Pull request by mvalipour**
Closes #12

### CODE REVIEW

File: lib/materialist/materialized_record.rb [hunk 1]
```diff
@@ -8,12 +9,20 @@ def self.included(base)
...
+            (via ? [via, key] : [key])
+              .inject(source_raw) do |res, path|
+                begin
+                  (res && res.body._links.include?(path)) ?
+                    res.send(path).show :
+                    (allow_nil ? nil : raise(ResourceNotFound))
+                rescue Routemaster::Errors::ResourceNotFound
+                  (allow_nil ? nil : raise(ResourceNotFound))
+                end
+              end
```
Comment (knaveofdiamonds) [inline lib/materialist/materialized_record.rb]: I find this section quite hard to parse - might be clearer with explicit `if` statements / extracted to its own method.

Comment (mvalipour) [reply lib/materialist/materialized_record.rb]: @robladbrook @knaveofdiamonds part of the reason this is left in it's convoluted form is to avoid adding any helper method here because any helpers here would be accessible through the materializer types.

It's not one of those things that you would come to read everyday, so I would leave it alone 😄
